webui/react/src/pages/TaskList.tsx

```diff
@@ -1,10 +1,33 @@
 import React from 'react';
 
 import Page from 'components/Page';
+import TaskTable from 'components/TasksTable';
+import { Commands, Notebooks, Shells, Tensorboards } from 'contexts/Commands';
+import { commandToTask } from 'utils/types';
 
 const TaskList: React.FC = () => {
+  const commands = Commands.useStateContext();
+  const notebooks = Notebooks.useStateContext();
+  const shells = Shells.useStateContext();
+  const tensorboards = Tensorboards.useStateContext();
+
+  const genericCommands = [
+    ...(commands.data || []),
+    ...(notebooks.data || []),
+    ...(shells.data || []),
+    ...(tensorboards.data || []),
+  ];
+
+  const loadedTasks = [
+    ...genericCommands.map(commandToTask),
+  ];
+
+  // TODO select and batch operation:
+  // https://ant.design/components/table/#components-table-demo-row-selection-and-operation
   return (
-    <Page title="Tasks" />
+    <Page title="Tasks">
+      <TaskTable tasks={loadedTasks} />
```

Comment:
suggestion: handle the empty states when there are no tasks due to there actually being no tasks, or the filter not matching anything. Two different copy for two different empty cases.

Reply:
suggestion: handle a loading state for when we when change pagination, refetch during sort changes, etc.

Reply:
If I'm getting the question right I think that IS handled here. loaded tasks can be 1. undefined 2. [] 3. non empty [] .. or I might have done this for the experiment list page let me check.

Reply:
it was missing from this one I added it.